src/control/tools/InputHandler.cpp

```diff
@@ -32,7 +31,7 @@ void InputHandler::resetShapeRecognizer() {
     // Does nothing here. Implemented in the extending classes
 }
 
-void InputHandler::createStroke(Point p) {
+void InputHandler::createStroke(Point p, const std::lock_guard<std::recursive_mutex>&) {
```

Comment:
Don't provide lock_guard's in the function header, certainly never as const reference. It may be useful, to assure, that a lock is acquired. But then it has to be either replaced with std::unique_lock (value semantic or non-const reference).